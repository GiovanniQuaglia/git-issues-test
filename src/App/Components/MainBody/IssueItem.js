import React from 'react';
import styled from 'styled-components';

const IssueItem = ({ issue }) => {
  const { title, html_url , number, user } = issue;
  return(
    <IssueWrapper>
      <Title href={html_url}>
        {title}
      </Title>
      <Details>
        {`#${number} created by ${user.login}`}
      </Details>
    </IssueWrapper>
  )
}

const IssueWrapper = styled.div`
  border-bottom: 1px solid lightgray;
  padding: 8px 0 0 16px;
  text-align: left;
  :hover {
    background-color: #F6F8FA;
    margin: 0 1px;
    padding: 8px 16px 0 16px;
  }
`;

const Title = styled.a`
  color: #24292e;
  font-weight: 600;
  text-decoration: none;
`;

const Details = styled.p`
  font-size: 12px;
  margin-top: 8px;
`;
export default IssueItem;