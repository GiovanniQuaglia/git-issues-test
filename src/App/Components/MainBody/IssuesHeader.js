import React from 'react';
import styled from 'styled-components';

const IssuesHeader = () => {
  return(
    <IssuesHeaderBody>
      Open
    </IssuesHeaderBody>
  )
}

const IssuesHeaderBody = styled.div`
  font-weight: 600;
  border-bottom: 1px solid lightgray;
  padding: 16px 0 16px 16px;
  text-align: left;
  background-color: #F6F8FA;
  color: #24292e;
`;

export default IssuesHeader;
