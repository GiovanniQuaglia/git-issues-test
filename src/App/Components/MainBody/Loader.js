import React from 'react';
import styled, {keyframes} from 'styled-components';

const Loader = () => {
  return(
    <Circle />
  )
}

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const Circle = styled.div`
  width: 60px;
  height: 60px;
  margin: 100px 0 100% 0;
  display: inline-block;
  animation: ${rotate} 1s linear infinite;
  background-color: #24292E;
  border-radius: 40px;

  ::before {
    content: '';
    width: 40px;
    height: 40px;
    background-color: #fff;
    z-index: 2;
    position: absolute;
    left: 10px;
    top: 10px;
    border-radius: 40px;
  }

  ::after {
    content: '';
    width: 10px;
    height: 60px;
    background-color: #fff;
    z-index: 2;
    position: absolute;
    left: 25px;
    top: 0px;
  }
`;

export default Loader;
