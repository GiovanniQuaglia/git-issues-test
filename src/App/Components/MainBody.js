import React, { Component } from 'react';
import styled from 'styled-components';
import axios from 'axios';

import Loader from './MainBody/Loader';

const API = 'https://api.github.com/repos/facebook/react/issues?page=';

class MainBody extends Component {
  constructor(props){
    super(props);
    this.state = {
      data: [],
      page: 1,
      isLoading: false,
      error: null,
    }
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    this.fetchData();
  }

  fetchData = async () => {
    try {
    const response = await axios.get(API + this.state.page);

    this.setState({
      data: response.data,
      isLoading: false,
    });
    console.log(response);
    }
    catch {
      this.setState({
        error: true,
        isLoading: false,
      })
    }
  }

  render() {
    const { isLoading, error } = this.state;

    return (
      <Wrapper>
        {isLoading &&
          <Loader />
        }
        {error &&
          <p>Molto male</p>
        }
      </Wrapper>
    );
  }
}


const Wrapper = styled.div`
  width: 100%;
  display: inline-block;
  border-top: 1px solid lightgray;
  margin-top: 32px;
  @media only screen and (min-width: 786px) {
    width: 90%;
    border: 1px solid lightgray;
    border-bottom: none;
    border-radius: 4px 4px 0 0;
  }
`;

export default MainBody;
