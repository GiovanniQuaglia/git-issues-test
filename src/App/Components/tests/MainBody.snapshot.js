import React from 'react';
import renderer from 'react-test-renderer';
import MainBody from '../MainBody';

describe('MainBody Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <MainBody />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
