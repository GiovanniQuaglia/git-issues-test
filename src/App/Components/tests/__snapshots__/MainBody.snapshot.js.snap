// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBody Snapshot renders 1`] = `
<div
  className="sc-EHOje dDliiN"
>
  <div
    className="sc-bdVaJa iWYoYO"
  />
</div>
`;
